FAQ.md

```diff
@@ -0,0 +1,80 @@
+
+**What are good resources to understand how ODEs can be solved?**<br>
+*Solving Ordinary Differential Equations I Nonstiff Problems* by Hairer et al.<br>
+[ODE solver selection in MatLab](https://blogs.mathworks.com/loren/2015/09/23/ode-solver-selection-in-matlab/)<br>
+
+**What are the ODE solvers available in this repo?**<br>
+
+- Adaptive-step:
+	- `dopri5` Runge-Kutta 4(5) [default].
+	- `adams` Adaptive-order implicit Adams.
+- Fixed-step:
+	- `euler` Euler's method.
+	- `midpoint` Midpoint method.
+	- `rk4` Fourth-order Runge-Kutta with 3/8 rule.
+	- `explicit_adams` Explicit Adams.
+	- `fixed_adams` Implicit Adams.
+
+
+**What are `NFE-F` and `NFE-B`?**<br>
+Number of function evaluations for forward and backward pass.
+
+**What are `rtol` and `atol`?**<br>
+They refer to relative `rtol` and absolute `atol` error tolerance.
+
+**What is the role of error tolerance in adaptive solvers?**<br>
+The basic idea is each adaptive solver can produce an error estimate of the current step, and if the error is greater than some tolerance, then the step is redone with a smaller step size, and this repeats until the error is smaller than the provided tolerance.
+[Error Tolerances for Variable-Step Solvers](https://www.mathworks.com/help/simulink/ug/types-of-solvers.html#f11-44943)
+
+**How is the error tolerance calculated?**<br>
+The error tolerance is [calculated]((https://github.com/rtqichen/torchdiffeq/blob/master/torchdiffeq/_impl/misc.py#L152)) as `atol + rtol * norm of current state`, where the norm being used is the infinity norm. 
+
+**Where is the code that computes the error tolerance?**<br>
+It is computed [here](https://github.com/rtqichen/torchdiffeq/blob/c4c9c61c939c630b9b88267aa56ddaaec319cb16/torchdiffeq/_impl/misc.py#L146)
+
+**How many states must a Neural ODE solver store during a forward pass with the adjoint method**<br>
+The number of states required to be stored in memory during a forward pass is solver dependent. For example, RK45 requires 6 intermediate states to be stored.
+
+**How many function evaluations are there per ODE step on adaptive solvers?**<br>
+
+- `dopri5`<br>
+	The `dopri5` ODE solver stores at least 6 evaluations of the ODE, then takes a step using a linear combination of them. The diagram below illustrates it: the evaluations marked with `o` are on the estimated path, the others with `x` are not. The first two are for selecting the initial step size.
+
+	0  1 |  2  3  4  5  6  7 |  8  9  10 12 13 14
+	o  x |  x  x  x  x  x  o |  x  x  x  x  x  o
+
+
+**How do I obtain evaluations on the estimated path when using an adaptive solver ?**<br>
+The argument `t` of `odeint` specifies what times should the ODE solver output.
+odeint(func, x0, t=torch.linspace(0, 1, 50))
+
+Note that the ODE solver will always integrate from `min t(0)` to `max t(1)`, and the intermediate values of `t` have no effect on how the ODE the solved. Computing intermediate values of t have a very small cost.
```

Comment:
and the intermediate values of `x(t)`

Intermediate values are computed using polynomial interpolation and have very small cost.